src/routes/profile.js

```diff
@@ -1,12 +1,60 @@
+let {createWriteStream} = require('fs');
+let debug = console.log.bind(console, '[routes/profile]');
+let eventToPromise = require('event-to-promise');
+let {exec} = require('mz/child_process');
 let express = require('express');
+let reboot = require('../fxos/reboot');
+let tempdir = require('../tempdir');
 
 let router = express.Router();
 
-function profile(req, res) {
-  // TODO
-  res.sendStatus(501);
+async function download(req, res) {
+  let {adb} = req;
+  let profile = await findProfile(adb);
```

Comment:
`adb` is only used with this single call, could probably just do `findProfile(req.adb)`.